### DISCUSSION TITLE
Question about mod-update.yml failing

### DISCUSSION THREAD

**Discussion by myrthos** [gethinode/hinode#471] [Help] [answered]
After I changed `npm ci` to `npm i` in `mod-update.yml` it gets a bit further, but it still fails. I changed the `mod-gallery` repo yesterday, but did not ran `npm run mod:update` on the main branch of my site yet, so `mod-update.yml` wants to do that and create a pull request branch. This is the error situation:

```
Create or update the pull request branch
  /usr/bin/git symbolic-ref HEAD --short
  main
  Working base is branch 'main'
  /usr/bin/git checkout --progress -B 17581f80-7872-41e1-8ba8-be1d1694865e HEAD --
  Switched to a new branch '17581f80-7872-41e1-8ba8-be1d1694865e'
  M	go.mod
  M	go.sum
  /usr/bin/git status --porcelain -unormal -- go.mod go.sum **/go.mod **/go.sum
   M go.mod
   M go.sum
  Uncommitted changes found. Adding a commit.
  /usr/bin/git add -- go.mod go.sum **/go.mod **/go.sum
  fatal: pathspec '**/go.mod' did not match any files
  /usr/bin/git -c author.name=myrthos -c author.email=myrthos@users.noreply.github.com -c committer.name=GitHub -c committer.email=noreply@github.com commit -m fix: update Hugo module dependencies
  On branch 17581f80-7872-41e1-8ba8-be1d1694865e
  Changes not staged for commit:
    (use "git add <file>..." to update what will be committed)
    (use "git restore <file>..." to discard changes in working directory)
  	modified:   go.mod
  	modified:   go.sum
  
  no changes added to commit (use "git add" and/or "git commit -a")
  Error: Unexpected error: 
```
 The error is in the middle `fatal: pathspec '**/go.mod' did not match any files`. At least I think it is, the word `fatal` sounds dramatic.

is this because I am not using the `exampleSite` folder?

**Comment by markdumay** [comment]
Yes, that's likely the cause. I'm struggling to get the `pathspec` right. I'd expect that `**/go.mod` would match `go.mod` in the repository root, but it doesn't. Likewise, I'd like to **only** include `go.mod` and `go.sum` recursively, but it should not panic when no files are found. Instead, the git PR should be ignored as there are no relevant files to merge.

If you don't use any modules in subfolders (such as `exampleSite`), you would have the remove the final two lines. I've added a comment to the code in `mod-template`. I would encourage to include an `exampleSite` though, as it would a) present a working, isolated demo of the shortcode / module and b) act as input for a build test.

**Reply by myrthos** [reply]
I have an exampleSite in mod-gallery, but not in myrthos-site. It is failing on the latter, not on mod-gallery.

**Reply by markdumay** [reply]
ok, clear! If you have any ideas on how to further streamline `mod-update.yml`, I'm all ears. :-)

**Reply by myrthos** [reply]
I have no experience in writing any GitHub actions, but would it be possible to check if exampleSite exists?

**Reply by markdumay** [reply]
Possibly, but I don't believe we should explore that path. Ideally, the glob pattern covers all scenarios. For example, we also want to support a scenario where `exampleSite` is renamed.

**Comment by markdumay** [comment]
Btw, I did not adjust `npm ci` to `npm i` for `mod-template` yet. I was expecting it would not be needed?

**Reply by myrthos** [reply]
I know, I changed it myself, It was needed because it failed every night.

**Reply by myrthos** [reply]
As extra info, it only fails after I manually run `npm update` on the command line and commit the changed json files. It causes a change related to `fsevents`, which results in a fail of `npm ci`

**Reply by markdumay** [reply]
Just to clarify: I have updated `mod-update.yml` for the `hinode` and `template` repositories. I did **not** update it for the `mod-template` repository. The revised `mod-update.yml` reverts back to `npm i` instead of `npm-ci`.

**Reply by myrthos** [reply]
Yes, I noticed that. My comments are off-topic and confusing. They are correct for my Hinode fork, but they do not apply to `mod-template`. I should learn to read better.

**Reply by markdumay** [reply, marked as answer]
No worries. For now I'd recommend to adjust `npm ci` to `npm i` in your action(s). Perhaps someone else has a better idea?
